Sources/SourceKitLSP/Swift/ExpandMacroCommand.swift

```diff
@@ -0,0 +1,74 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import LanguageServerProtocol
+import SourceKitD
+
+public struct ExpandMacroCommand: RefactorCommand {
+  public static let identifier: String = "expand.macro.command"
+
+  /// The name of this refactoring action.
+  public var title = "Expand Macro"
+
+  /// The sourcekitd identifier of the refactoring action.
+  public var actionString = "source.refactoring.kind.expand.macro"
+
+  /// The range to expand.
+  public var positionRange: Range<Position>
+
+  /// The text document related to the refactoring action.
+  public var textDocument: TextDocumentIdentifier
+
+  public init(positionRange: Range<Position>, textDocument: TextDocumentIdentifier) {
+    self.positionRange = positionRange
+    self.textDocument = textDocument
+  }
+
+  public init?(fromLSPDictionary dictionary: [String: LSPAny]) {
+    guard case .dictionary(let documentDict)? = dictionary[CodingKeys.textDocument.stringValue],
+      case .string(let title)? = dictionary[CodingKeys.title.stringValue],
+      case .string(let actionString)? = dictionary[CodingKeys.actionString.stringValue],
+      case .dictionary(let rangeDict)? = dictionary[CodingKeys.positionRange.stringValue]
+    else {
+      return nil
+    }
+    guard let positionRange = Range<Position>(fromLSPDictionary: rangeDict),
+      let textDocument = TextDocumentIdentifier(fromLSPDictionary: documentDict)
+    else {
+      return nil
+    }
+
+    self.init(
+      title: title,
+      actionString: actionString,
+      positionRange: positionRange,
+      textDocument: textDocument
+    )
+  }
+
+  public init(title: String, actionString: String, positionRange: Range<Position>, textDocument: TextDocumentIdentifier)
+  {
```

Comment:
Nitpick, I prefer to break this on the parameters.

```suggestion
  public init(
    title: String, 
    actionString: String, 
    positionRange: Range<Position>, 
    textDocument: TextDocumentIdentifier
  ) {
```